### PR TITLE
Add manifest classes

### PR DESCRIPTION
**Pull request by erikzhang**
Related to https://github.com/neo-project/neo/issues/2571#issuecomment-897654986

### CODE REVIEW

File: src/Neo.SmartContract.Framework/Services/ContractMethodDescriptor.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace Neo.SmartContract.Framework.Services
+{
+    public struct ContractMethodDescriptor
+    {
+        public string Name;
+        public ContractParameterDefinition[] Parameters;
+        public ContractParameterType ReturnType;
+        public int Offset;
+        public bool Safe;
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Services/ContractMethodDescriptor.cs:9]: It doesn't have all of these values

https://github.com/neo-project/neo/blob/c8604f41a39e8877621469304381b94089be5bc7/src/neo/SmartContract/Manifest/ContractMethodDescriptor.cs#L51-L55

Comment (erikzhang) [reply src/Neo.SmartContract.Framework/Services/ContractMethodDescriptor.cs:9]: The other fields are in the base class.

**Comment by shargon** [comment]
@superboyiii did you test it? something it's failing in my tests

**Comment by erikzhang** [comment]
Go?

**Comment by superboyiii** [comment]
> @superboyiii did you test it? something it's failing in my tests

Let me try

**Comment by shargon** [comment]
@superboyiii I mean using the change, please check https://github.com/neo-project/neo-devpack-dotnet/pull/680

**Comment by superboyiii** [comment]
> @superboyiii I mean using the change, please check #680

UT seems already works on branch `add-manifest`

**Comment by superboyiii** [comment]
@erikzhang  https://github.com/neo-project/neo-devpack-dotnet/pull/680 is good to merge to this.

**Comment by superboyiii** [comment]
```
namespace TestNetFee
{
    [ContractPermission("0x42e54382e86dcdca09e0da8bb67e2fac4d498744", "test")]
    [SupportedStandards("NEP-17")]
    [ManifestExtra("Author", "Neo")]
    [ManifestExtra("Email", "dev@neo.org")]
    [ManifestExtra("Description", "This is a contract example")]
    [ContractSourceCode("https://github.com/neo-project/neo")]
    public class TestNetFee : SmartContract
```

```
        public static string TestPermission()
        {
            Contract c = ContractManagement.GetContract(Runtime.ExecutingScriptHash);
            return c.Manifest.Permissions.ToString();
        }

        public static string TestSupportedStandards()
        {
            Contract c = ContractManagement.GetContract(Runtime.ExecutingScriptHash);
            return c.Manifest.SupportedStandards.ToString();
        }
```

```
neo> invoke 0x5aade60da2ae1b83a9715a2a669963a298501fd2 testPermission []
Invoking script with: 'wh8MDnRlc3RQZXJtaXNzaW9uDBTSH1CYomOZZipacamDG66iDeatWkFifVtS'
VM State: FAULT
Gas Consumed: 0.0302385
Result Stack: []
Error: The value -1 is out of range.
neo> invoke 0x5aade60da2ae1b83a9715a2a669963a298501fd2 testSupportedStandards []
Invoking script with: 'wh8MFnRlc3RTdXBwb3J0ZWRTdGFuZGFyZHMMFNIfUJiiY5lmKlpxqYMbrqIN5q1aQWJ9W1I='
VM State: FAULT
Gas Consumed: 0.0302385
Result Stack: []
Error: The value -1 is out of range.
```

**Comment by superboyiii** [comment]
```
        public static object TestPermission()
        {
            Contract c = ContractManagement.GetContract(Runtime.ExecutingScriptHash);
            return c.Manifest.Permissions;
        }
```
Seems vm doesn't support string[].toString(), but return object straightly works well.
```
neo> invoke 0x5aade60da2ae1b83a9715a2a669963a298501fd2 testPermission
Invoking script with: 'wh8MDnRlc3RQZXJtaXNzaW9uDBTSH1CYomOZZipacamDG66iDeatWkFifVtS'
VM State: HALT
Gas Consumed: 0.0301995
Result Stack: [{"type":"Array","value":[{"type":"Struct","value":[{"type":"ByteString","value":"RIdJTawvfraL2uAJys1t6IJD5UI="},{"type":"Array","value":[{"type":"ByteString","value":"dGVzdA=="}]}]},{"type":"Struct","value":[{"type":"ByteString","value":"wO85zuDk6SXGwqBqeeFEDdhvzqw="},{"type":"Array","value":[{"type":"ByteString","value":"aXRvYQ=="},{"type":"ByteString","value":"anNvbkRlc2VyaWFsaXpl"}]}]},{"type":"Struct","value":[{"type":"ByteString","value":"WIcXEX4KqBByr6tx0t2J/nxLkv4="},{"type":"Array","value":[{"type":"ByteString","value":"cmVxdWVzdA=="}]}]},{"type":"Struct","value":[{"type":"ByteString","value":"/aP6Q0bqUyolj8SX3a3bZDfJ/f8="},{"type":"Array","value":[{"type":"ByteString","value":"ZGVzdHJveQ=="},{"type":"ByteString","value":"Z2V0Q29udHJhY3Q="},{"type":"ByteString","value":"dXBkYXRl"}]}]}]}]
```

**Comment by superboyiii** [comment]
Test Passed
@shargon Review again?

**Review by superboyiii — APPROVED** [review]

File: tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_Update.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@ public class Contract_Update : SmartContract
     {
         public static string NewContract()
         {
+            Policy.GetFeePerByte();
```
Comment (shargon) [inline tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_Update.cs:10]: This is needed?

Comment (shargon) [reply tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_Update.cs:10]: It seems needed, but why @erikzhang ?

Comment (erikzhang) [reply tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_Update.cs:10]: It should be fixed by neo-project/neo#2611

**Review by shargon — APPROVED** [review]

**Pull request merged**